### ISSUE TITLE
IMAP not working with outlook.office365.com (due to changements in the Office 365 service ?)

### ISSUE THREAD

**Issue by hasse-basse** [Dolibarr/dolibarr#22517]
### Bug

Trying to use "Collecteur de courrier électronique" but it does not work, I have the message "Failed to open IMAP connection {outlook.office365.com:993/service=imap/ssl/novalidate-cert}INBOX".

I put the question why to a FB group and had the answer "Deprecation of Basic authentication in Exchange Online". In other words, Microsft has made some changes in the service. Pls see the below link link.

I assume that some changes has to be done in the module?

https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online?fbclid=IwAR1pTH5Zkbk7Lc5PutzNLe0ZXMHRbrBcA6KLpGirMR9Zs7LyTiX6M20Y-uA

### Environment Version

16

### Environment OS

Debian 11

### Environment Web server

Apache2

### Environment PHP

7.4

**Comment by priojk** [comment]
Well, what Microsoft did: They disabled the possibility to login via username and password, even if you set an application specific password (which was an option up to now). The only way left is login via OAuth2, which is quite complex and cumbersome (and therefore expensive) to implement. Don't know if a developer needs a formal certification from Microsoft for the implementation, Google requires this and the costs are five digit upwards (an approved company will do an audit of the source code of the system that wants to connect).

In the end, Microsoft and Google (and probably some other major players as well) abandon step by step the open, interoperable aspects of the original internet. Previously, I as the owner of my system could decide, to which other systems I want to connect and use open protocols like IMAP. Now, there get (semi-)walled gardens created. If you now want to offer something universal (like connection to IMAP mailboxes), you need to get agreement by all the owners of the new walled gardens.

Long story short: This is not a bug of Dolibarr. Everything works according to the IMAP specification. The bug is rather with Microsoft, who removed parts of the standard IMAP functionality from their service offerings.

**Comment by hasse-basse** [comment]
Oh dear, it's just to give it up then? Thanks for the answer. So what will the next step be? That they take away the possibility to use their SMTP server as well?
